general/mem_manager.hpp

```diff
@@ -175,14 +175,28 @@ class Memory
    /// Copy constructor: default.
    Memory(const Memory &orig) = default;
 
-   /// Move constructor: default.
-   Memory(Memory &&orig) = default;
+   /// Move constructor, takes ownership when possible and invalidates @a orig.
```

Comment:
What does it mean "when possible"?

Reply:
"when `@a orig` has ownership", would you prefer that in the documentation?

Reply:
Maybe something like
```c++
/// Move constructor. Sets the pointers and associated ownership of validity
/// flags of @a *this to those of @a other. Invalidates @a other.
```